bentoml/_internal/bento/bento.py

```diff
@@ -312,23 +408,65 @@ def from_Bento(bento: Bento) -> "SysPathBento":
     def path(self) -> str:
         return self._fs.getsyspath("/")
 
+    @inject
     def save(
-        self, bento_store: "BentoStore" = Provide[BentoMLContainer.bento_store]
+        self, *, bento_store: "BentoStore" = Provide[BentoMLContainer.bento_store], force: bool = False
     ) -> "SysPathBento":
         if not self.validate():
             logger.warning(f"Failed to create Bento for {self.tag}, not saving.")
             raise BentoMLException("Failed to save Bento because it was invalid")
 
-        with bento_store.register(self.tag) as bento_path:
-            os.rmdir(bento_path)
+        with bento_store.register(self.tag, force=force) as bento_path:
+            shutil.rmtree(bento_path)
             shutil.move(self._fs.getsyspath("/"), bento_path)
             self._fs.close()
             self._fs = fs.open_fs(bento_path)
 
         return self
 
 
-class BentoStore(Store[SysPathBento]):
+class BentoStore(RemoteStore[SysPathBento]):
+    @inject
+    def pull(self, tag: t.Union[str, Tag], *, force: bool = False,
+             model_store: "ModelStore" = Provide[BentoMLContainer.model_store]) -> RemoteItem:
+        if not force:
+            try:
+                bento = self.get(tag)
+                logger.info(f"Bento {tag} already exists locally, skipping pull")
+                return bento
+            except NotFound:
```

Comment:
Could we do a delete here when force is True? instead of adding it to all the store APIs?

The reason is in almost all other cases, `force` can be a dangerous behavior and we would never recommend users doing that. Although having it in the API parameter is tempting. I slightly prefer asking the user do a explicitly delete and then add, if they actually need the `force` behavior